qa/rolling-upgrade/src/test/java/org/opensearch/upgrades/ClusterStateIT.java

```diff
@@ -8,30 +8,369 @@
+//            Settings are required to wiped after Tests, so Settings are not present after rolling upgrade starts
+//            verifySettingsInClusterState();
+        }
+    }
+
     public void testTemplateMetadataUpgrades() throws Exception {
         if (CLUSTER_TYPE == ClusterType.OLD) {
-            String templateName = "my_template";
-            Request putIndexTemplate = new Request("PUT", "_template/" + templateName);
-            putIndexTemplate.setJsonEntity("{\"index_patterns\": [\"pattern-1\", \"log-*\"]}");
-            client().performRequest(putIndexTemplate);
-            verifyTemplateMetadataInClusterState();
+            createIndexTemplate();
+            Request request = new Request("GET", "_cluster/state/metadata");
+            Response response = client().performRequest(request);
+            assertOK(response);
+
+            verifyTemplateMetadataInClusterState(response);
         } else {
-            verifyTemplateMetadataInClusterState();
+            Request request = new Request("GET", "_cluster/state/metadata");
+            Response response = client().performRequest(request);
+            assertOK(response);
+
+            verifyTemplateMetadataInClusterState(response);
+        }
+    }
+
+    private static void createIndexTemplate() throws Exception {
+        Request putIndexTemplate = new Request("PUT", "_template/" + templateName);
+        String indexTemplateJson = "{\n" +
+            "  \"index_patterns\": [\"pattern-*\", \"log-*\"],\n" +
+            "  \"settings\": {\n" +
+            "    \"number_of_shards\": 3,\n" +
+            "    \"number_of_replicas\": 1,\n" +
+            "    \"index.refresh_interval\": \"5s\"\n" +
+            "  },\n" +
+            "  \"mappings\": {\n" +
+            "    \"properties\": {\n" +
+            "      \"timestamp\": {\n" +
+            "        \"type\": \"date\"\n" +
+            "      },\n" +
+            "      \"message\": {\n" +
+            "        \"type\": \"text\",\n" +
+            "        \"fields\": {\n" +
+            "          \"keyword\": {\n" +
+            "            \"type\": \"keyword\",\n" +
+            "            \"ignore_above\": 256\n" +
+            "          }\n" +
+            "        }\n" +
+            "      },\n" +
+            "      \"level\": {\n" +
+            "        \"type\": \"keyword\"\n" +
+            "      }\n" +
+            "    }\n" +
+            "  },\n" +
+            "  \"aliases\": {\n" +
+            "    \"mydata\": {}\n" +
+            "  },\n" +
+            "  \"version\": 1\n" +
+            "}";
+
+        putIndexTemplate.setJsonEntity(indexTemplateJson);
+        client().performRequest(putIndexTemplate);
+    }
+
+    private static void createComponentTemplate() throws Exception {
+        Request putComponentTemplate = new Request("PUT", "_component_template/" + componentTemplateName);
+        String componentTemplateJson = "{\n" +
```

Comment:
There is one limitation with these ITs as they will not cover sub fields added future.

Reply:
Agreed, I don't think there is a framework we can leverage right now to cover all kinds of fields that are added in the future.
Do you think this is a valid feature request we can create?